src/Time/Units.hs

```diff
@@ -85,3 +88,11 @@ convertUnit (Time ratNat) = Time $ ratNat * (ratVal $ divRat (Proxy @unitFrom) (
 -- | Extracts integer part of time unit.
 intPart :: Time unit -> Natural
 intPart (Time n) = numerator n `div` denominator n
+
+-- | Convenient version of 'Control.Concurrent.threadDelay' which takes
+-- any time-unit and operates in any MonadIO.
+threadDelay :: forall unit m .
+               (KnownRat unit, KnownRat (DivRat unit MicroSecondUnit), MonadIO m)
```

Comment:
We need to do something to not specify `KnownRat` for `DivRat`... I don't know how to do it, need to think more.

Reply:
This sort of redundant constraints is solvable via typechecker plugins such as https://github.com/clash-lang/ghc-typelits-natnormalise, we might need our own for `KnownRat`.

Alternatively, we can use `Dict` and `unsafeCoerce` to manipulate dictionaries manually and produce desired `KnownRat` constraints, i.e.

```
withDivRat :: (KnownRat a, KnownRat b) => (KnownRat (DivRat a b) => r) -> r
withDivRat = ... -- unsafeCoerce magic
```